contentcuration/contentcuration/frontend/channelList/views/ChannelSet/ChannelSetModal.vue

```diff
@@ -350,44 +383,54 @@
       },
       confirmCancel() {
         if (this.isNew) {
-          return this.deleteChannelSet(this.channelSet).then(this.close);
+          if (this.channelSet && this.channelSet.id) {
+            return this.deleteChannelSet(this.channelSet).then(this.close);
+          } else {
+            this.close();
+          }
+        } else {
+          this.close();
         }
-        this.close();
       },
+
       close() {
         this.changed = false;
         this.showUnsavedDialog = false;
         this.diffTracker = {};
         this.$router.push({ name: RouteNames.CHANNEL_SETS });
       },
+
       verifyChannelSet(channelSetId) {
-        return new Promise((resolve, reject) => {
-          // Check if we already have the channel locally
-          if (this.getChannelSet(channelSetId)) {
-            this.setup();
-            resolve();
-            return;
-          }
+        if (!channelSetId || channelSetId === 'new') {
+          return Promise.resolve();
+        }
+
+        if (this.getChannelSet(channelSetId)) {
+          this.setup();
+          return Promise.resolve();
+        }
 
-          // If not, try to load the channel
-          this.loadChannelSet(channelSetId).then(channelset => {
-            // Did our fetch return any channels, then we have a channel!
-            if (channelset) {
+        return this.loadChannelSet(channelSetId)
+          .then(channelSet => {
+            if (channelSet) {
               this.setup();
-              resolve();
               return;
             }
-            // If not, reject!
+
+            this.$router.replace({ name: 'NEW_CHANNEL_SET' });
 
             this.$store.dispatch('errors/handleGenericError', {
               errorType: ErrorTypes.PAGE_NOT_FOUND,
               errorText: this.$tr('collectionErrorText'),
             });
-            reject();
+          })
+          .catch(error => {
+            console.error('Error loading collection:', error);
```

Comment:
With the `loadChannelSet` error being suppressed, I don't think this will be useful

https://github.com/learningequality/studio/blob/dbb5a639508ea483c9563958a75a885b075886b9/contentcuration/contentcuration/frontend/channelList/vuex/channelSet/actions.js#L12-L21